paperqa/agents/models.py

```diff
@@ -0,0 +1,535 @@
+    docs_index_embedding_config: dict[str, Any] | None = None
+    parsing_configuration: ParsingConfiguration = ParsingConfiguration()
+    embedding: str = "text-embedding-3-small"
+    # concurrent number of summary calls to use inside Doc object
+    max_concurrent: int = 20
+    temperature: float = 0.0
+    summary_temperature: float = 0.0
+    # at what size should we start using adoc_match?
+    adoc_match_threshold: int = 500
+    # Should we filter out "Extra Background Information" citations
+    # which come from pre-step in paper-qa algorithm
+    filter_extra_background: bool = True
+    # provides post-hoc linkage of request to a docs object
+    # NOTE: this isn't a unique field, on the user to keep straight
+    _docs_name: str | None = PrivateAttr(default=None)
+
+    # strict validation for now
+    model_config = ConfigDict(extra="forbid")
+
+    @computed_field  # type: ignore[misc]
+    @property
+    def docs_name(self) -> str | None:
+        return self._docs_name
+
+    @model_validator(mode="after")
+    def llm_models_must_match(self) -> QueryRequest:
+        llm = llm_model_factory(self.llm)
+        summary_llm = llm_model_factory(self.summary_llm)
+        if type(llm) is not type(summary_llm):
+            raise MismatchedModelsError(
+                f"Answer LLM and summary LLM types must match: {type(llm)} != {type(summary_llm)}"
+            )
+        return self
+
+    @field_validator("prompts")
+    def update_prompts(
+        cls,  # noqa: N805
+        v: PromptCollection,
+        info: ValidationInfo,
+    ) -> PromptCollection:
+        values = info.data
+        if values["named_prompt"] is not None:
+            if values["named_prompt"] not in STATIC_PROMPTS:
+                raise ValueError(
+                    f"Named prompt {values['named_prompt']} not in {list(STATIC_PROMPTS.keys())}"
+                )
+            v = STATIC_PROMPTS[values["named_prompt"]]
+        if values["summary_llm"] == "none":
+            v.skip_summary = True
+            # for simplicity (it is not used anywhere)
+            # so that Docs doesn't break when we don't have a summary_llm
+            values["summary_llm"] = "gpt-4o-mini"
+        return v
+
+    def set_docs_name(self, docs_name: str) -> None:
+        """Set the internal docs name for tracking."""
+        self._docs_name = docs_name
+
+    @staticmethod
+    def get_index_name(
+        paper_directory: str | os.PathLike,
+        embedding: str,
+        parsing_configuration: ParsingConfiguration,
+    ) -> str:
+
+        index_fields = "|".join(
+            [
+                str(paper_directory),
+                embedding,
+                str(parsing_configuration.chunksize),
+                str(parsing_configuration.overlap),
+                parsing_configuration.chunking_algorithm,
+            ]
+        )
+
+        return (
+            f"pqa_index_{hashlib.md5(index_fields.encode()).hexdigest()}"  # noqa: S324
```

Comment:
Can you move this encode functionality to a free function in `utils` or the module level here?